packages/sanity/src/core/store/_legacy/history/useTimelineStore.ts

```diff
@@ -0,0 +1,209 @@
+         * Until then, this workaround waits until the next call stack before calling `setRange`.
+         */
+        setTimeout(() => {
+          innerController.setRange(since || null, rev || null)
+          timelineController$.next(innerController)
+        }, 0)
+      }
+    }
+    controller.resume()
+    return () => controller.suspend()
+  }, [rev, since, controller, timelineController$])
+
+  /**
+   * Fetch document snapshots and update the mutable controller.
+   * Unsubscribes on clean up, preventing double fetches in strict mode.
+   */
+  useEffect(() => {
+    if (!snapshotsSubscriptionRef.current) {
+      snapshotsSubscriptionRef.current = remoteSnapshots(
+        client,
+        {draftId: `drafts.${documentId}`, publishedId: documentId},
+        documentType
+      ).subscribe((ev: RemoteSnapshotVersionEvent) => {
+        controller.handleRemoteMutation(ev)
+      })
+    }
+    return () => {
+      if (snapshotsSubscriptionRef.current) {
+        snapshotsSubscriptionRef.current.unsubscribe()
+        snapshotsSubscriptionRef.current = null
+      }
+    }
+  }, [client, controller, documentId, documentType])
+
+  const timelineStore = useMemo(() => {
+    return {
+      findRangeForRev: (chunk: Chunk) => controller.findRangeForNewRev(chunk),
+      findRangeForSince: (chunk: Chunk) => controller.findRangeForNewSince(chunk),
+      loadMore: () => {
+        controller.setLoadMore(true)
+        timelineStateRef.current.isLoading = true
+      },
+      getSnapshot: () => timelineStateRef.current,
+      subscribe: (callback: () => void) => {
+        const subscription = timelineController$
+          .pipe(
+            // Manually stop loading transactions in TimelineController, otherwise transaction history
+            // will continue to be fetched – even if unwanted.
+            tap((innerController) => innerController.setLoadMore(false)),
+            map((innerController) => ({
+              chunks: innerController.timeline.mapChunks((c) => c),
+              diff: innerController.sinceTime ? innerController.currentObjectDiff() : null,
+              isLoading: false,
+              hasMoreChunks: !innerController.timeline.reachedEarliestEntry,
+              onOlderRevision: innerController.onOlderRevision(),
+              realRevChunk: innerController.realRevChunk,
+              revTime: innerController.revTime,
+              selectionState: innerController.selectionState,
+              sinceAttributes: innerController.sinceAttributes(),
+              sinceTime: innerController.sinceTime,
+              timelineDisplayed: innerController.displayed(),
+              timelineReady: !['invalid', 'loading'].includes(innerController.selectionState),
+            })),
+            // Only emit (and in turn, re-render) when values have changed
+            distinctUntilChanged(deepEquals),
+            tap((timelineState) => {
+              timelineStateRef.current = timelineState
```

Comment:
Since this happens before `catchError`, the current state won't be set to `INITIAL_TIMELINE_STATE` in case of an error - is this the behavior we want?

Reply:
Cripes, it certainly isn't – that's been updated!